File: front-end/src/app/service-models/app/v1/client/client-employee-c-t-c-detail-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { ClientEmployeePayrollComponentSM } from './client-employee-payroll-component-s-m';

export class ClientEmployeeCTCDetailSM extends CoinManagementServiceModelBase<number> {
    ctcAmount!: number;
    currencyCode!: string;
    startDateUTC!: Date;
    endDateUTC?: Date;
    currentlyActive!: boolean;
    clientUserId!: number;
    clientEmployeePayrollComponents!: Array<ClientEmployeePayrollComponentSM>;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientEmployeeCTCDetailSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using System.Collections.Generic;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientEmployeeCTCDetailSM : SiffrumPayrollServiceModelBase<int>
    {
        public float CtcAmount { get; set; }

        public string CurrencyCode { get; set; }

        public DateTime StartDateUTC { get; set; }

        public DateTime? EndDateUTC { get; set; }

        public bool CurrentlyActive { get; set; }

        public int ClientUserId { get; set; }

        //public int ClientCompanyDetailId { get; set; }

        public List<ClientEmployeePayrollComponentSM> ClientEmployeePayrollComponents { get; set; }

    }
}
